operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/AbstractDependentResource.java

```diff
@@ -22,16 +20,16 @@
     implements DependentResource<R, P> {
   private static final Logger log = LoggerFactory.getLogger(AbstractDependentResource.class);
 
-  protected final boolean creatable = this instanceof Creator;
-  protected final boolean updatable = this instanceof Updater;
-  protected final boolean bulk = this instanceof BulkDependentResource;
+  private final boolean creatable = this instanceof Creator;
+  private final boolean updatable = this instanceof Updater;
+  private final boolean bulk = this instanceof BulkDependentResource;
 
   protected Creator<R, P> creator;
   protected Updater<R, P> updater;
   protected BulkDependentResource<R, P> bulkDependentResource;
   private boolean returnEventSource = true;
-
-  protected List<ResourceDiscriminator<R, P>> resourceDiscriminator = new ArrayList<>(1);
+  private ResourceDiscriminator<R, P> resourceDiscriminator;
+  private int currentCount;
```

Comment:
Should we address this global variable in this PR, or  rather should make a separate one?

Reply:
I would say in a different PR but we can do it here if it makes more sense.

Reply:
sure np